### PR TITLE
Component Names #7634

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/content/ComponentNameResolverImpl.java  6
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/ComponentJson.java  2
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/page/region/GetComponentByKeyCommand.java  5
         

Complexity decreasing per file
==============================
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/TextComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/ImageComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/LayoutComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/PartComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/FragmentComponentJson.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/enonic/xp/pullRequest?prid=4673962&bid=15640881)

File: modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/postprocess/instruction/ComponentInstruction.java [hunk 1]
```diff
@@ -18,14 +19,15 @@
...
 import com.enonic.xp.region.Component;
-import com.enonic.xp.region.ComponentName;
 import com.enonic.xp.region.ComponentPath;
 import com.enonic.xp.region.ComponentService;
 import com.enonic.xp.region.LayoutComponent;
 import com.enonic.xp.region.LayoutRegions;
 import com.enonic.xp.trace.Trace;
 import com.enonic.xp.trace.Tracer;
 
+import static org.apache.commons.lang.StringUtils.substringAfter;
```
Comment (rymsha) [inline modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/postprocess/instruction/ComponentInstruction.java]: Don't use commons-lang #7716

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/page/region/GetComponentByKeyCommand.java  5
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/content/ComponentNameResolverImpl.java  6
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/ComponentJson.java  2
         

Complexity decreasing per file
==============================
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/PartComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/LayoutComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/TextComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/FragmentComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/ImageComponentJson.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/enonic/xp/pullRequest?prid=4673962&bid=15640881)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/page/region/GetComponentByKeyCommand.java  5
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/content/ComponentNameResolverImpl.java  6
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/ComponentJson.java  2
         

Complexity decreasing per file
==============================
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/LayoutComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/ImageComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/PartComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/FragmentComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/TextComponentJson.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/enonic/xp/pullRequest?prid=4673962&bid=15640881)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/content/ComponentNameResolverImpl.java  6
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/ComponentJson.java  2
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/page/region/GetComponentByKeyCommand.java  5
         

Complexity decreasing per file
==============================
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/TextComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/PartComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/ImageComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/LayoutComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/FragmentComponentJson.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/enonic/xp/pullRequest?prid=4673962&bid=15640881)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/page/region/GetComponentByKeyCommand.java  5
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/content/ComponentNameResolverImpl.java  6
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/ComponentJson.java  2
         

Complexity decreasing per file
==============================
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/LayoutComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/PartComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/TextComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/FragmentComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/ImageComponentJson.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/enonic/xp/pullRequest?prid=4673962&bid=15640881)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/ComponentJson.java  2
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/page/region/GetComponentByKeyCommand.java  5
- modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/rest/resource/content/ComponentNameResolverImpl.java  6
         

Complexity decreasing per file
==============================
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/ImageComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/TextComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/LayoutComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/PartComponentJson.java  -2
+ modules/admin/admin-impl/src/main/java/com/enonic/xp/admin/impl/json/content/page/region/FragmentComponentJson.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=4673962&bid=15640881)

**Pull request closed without merging**